File: services/dotnet-service/Controllers/PlatformController.cs
using Microsoft.AspNetCore.Mvc;

[ApiController]
[Route("/")]
public class PlatformController : ControllerBase
{
    private static readonly Random Random = new();

    [HttpGet("health")]
    public IActionResult Health()
    {
        return Ok(new { status = "UP" });
    }

    [HttpGet("process")]
    public IActionResult Process()
    {
        int delay = Random.Next(300, 2000);
        Thread.Sleep(delay);

        if (Random.Next(10) < 2)
        {
            return StatusCode(500, new { error = "Simulated processing failure" });
        }

        return Ok(new
        {
            message = "Processed successfully",
            processing_time_ms = delay
        });
    }

    [HttpGet("error")]
    public IActionResult Error()
    {
        return StatusCode(500, new { error = "Intentional error triggered" });
    }
}

File: services/java-service/src/main/java/com/platform/java_service/controller/PlatformController.java
package com.platform.java_service.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Random;

@RestController
public class PlatformController {

    private final Random random = new Random();

    @GetMapping("/health")
    public String health() {
        return "UP";
    }

    @GetMapping("/process")
    public String process() throws InterruptedException {
        int delay = random.nextInt(2000);
        Thread.sleep(delay);

        if (random.nextInt(10) < 2) {
            throw new RuntimeException("Simulated processing failure");
        }

        return "Processed successfully in " + delay + " ms";
    }

    @GetMapping("/error")
    public String error() {
        throw new RuntimeException("Intentional error triggered");
    }
}
